### PR TITLE
Improve JDK detection for bootstrapping

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: utils/ort/src/test/kotlin/JavaBootstrapperTest.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.utils.ort
+
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.shouldBe
+
+class JavaBootstrapperTest : StringSpec({
+    "The current JDK should be detected as a JDK" {
```
Comment (mnonnenmacher) [inline utils/ort/src/test/kotlin/JavaBootstrapperTest.kt:26]: The test name is confusing, I don't understand if this is a test for `isRunningOnJdk` or for `Environment.JAVA_VERSION`.

File: utils/ort/src/main/kotlin/JavaBootstrapper.kt [hunk 2]
```diff
@@ -58,7 +59,11 @@ object JavaBootstrapper {
     fun isRunningOnJdk(version: String): Boolean {
         val requestedVersion = Semver.coerce(version)
         val runningVersion = Semver.coerce(Environment.JAVA_VERSION)
-        return requestedVersion == runningVersion
+        if (requestedVersion != runningVersion) return false
+
+        val javaHome = System.getProperty("java.home") ?: return false
+        val javac = File(javaHome).resolve("bin").resolve("javac")
+        return Os.resolveWindowsExecutable(javac) != null
```
Comment (mnonnenmacher) [inline utils/ort/src/main/kotlin/JavaBootstrapper.kt:66]: I was wondering why this is not wrapped with some `if (Os.isWindows)` check, but looking at the function it seems that on Linux it just checks if the file exists. Maybe the function should be renamed to just `resolveExecutable`, reading it here like this is unexpected.

File: plugins/package-managers/gradle/src/main/kotlin/Gradle.kt [hunk 3]
```diff
@@ -220,12 +236,30 @@ class Gradle(
...
+                                    }.getOrNull()
+                            } else {
+                                null
+                            }
+                        } ?: options[OPTION_JAVA_HOME]?.let { File(it) }
+
+                        javaHome?.also {
+                            logger.info { "Setting Java home for project analysis to '$it'." }
+                            setJavaHome(it)
+                        }
```
Comment (mnonnenmacher) [inline plugins/package-managers/gradle/src/main/kotlin/Gradle.kt:262]: Maybe:

```kotlin
val javaHome = options[OPTION_JAVA_VERSION]
    ?.takeIf { !JavaBootstrapper.isRunningOnJdk(it) }
    ?.let {
        JavaBootstrapper.installJdk("TEMURIN", it)
            .onFailure { e -> issues += createAndLogIssue(managerName, e.collectMessages()) }
            .getOrNull()
    } ?: options[OPTION_JAVA_HOME]?.let { File(it) }
```

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: utils/ort/src/test/kotlin/JavaBootstrapperTest.kt (hunk 1 above)
Comment (sschuberth) [reply utils/ort/src/test/kotlin/JavaBootstrapperTest.kt:26]: The point of this test is that a JDK should be detected as a JDK, and not as a JRE. As tests are always run with a JDK, the test should succeed.

Comment (mnonnenmacher) [reply utils/ort/src/test/kotlin/JavaBootstrapperTest.kt]: Ok, that clarifies it for me. Maybe mentioning the function name in the test name would have helped, but that's not so important.

**Review by mnonnenmacher — APPROVED** [review]

**Pull request merged**